swig/lightgbmlib.i

```diff
@@ -133,6 +133,15 @@
           int size;
   };
 
+  void check_exception(JNIEnv *jenv) {
+    jboolean flag = jenv->ExceptionCheck();
+    if (flag) {
+      jthrowable jerror = jenv->ExceptionOccurred();
+      jenv->ExceptionClear();
+      jenv->Throw(jerror);
```

Comment:
I'm not sure this does what you want. jenv->Throw() will not throw, but rather mark the Java stack to throw a Java exception. if you call another Java method from the JNI side it will crash (at least in my local tests)

Reply:
I think we should have `LGBM_DatasetCreateFromCSRSpark` exit early and return `-1` in addition to having the java side throw.

Reply:
done, I also validated by adding:

```
jenv->GetObjectArrayElement(arrayOfSparseVector, -1);
```

prior to check_exception, and the program seemed to fail gracefully with:

```
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: -1
	at com.microsoft.ml.lightgbm.lightgbmlibJNI.LGBM_DatasetCreateFromCSRSpark(Native Method)
	at com.microsoft.ml.lightgbm.lightgbmlib.LGBM_DatasetCreateFromCSRSpark(lightgbmlib.java:290)
	at program.generateSparseDataset(program.java:22)
	at program.main(program.java:54)
```

Reply:
and of course, I also validated that the warnings are still fixed